CI/CD initial workflow

bunbry opened the pull request:
Workflows implementing steps for building, testing, and deployment. The build and deploy steps consist of reusable workflows dependent on aktin/aktin-github-scripts#2

bunbry committed:
ci: remove old GitHub workflow

bunbry committed:
ci: add Workflow with build job and deploy job

bunbry committed:
ci: account for new inputs of deploy workflow

The inputs package-name and package-version were added as mandatory inputs of the debian-deploy.yml workflow.

bunbry committed:
ci: add test job (only installation implemented)

bunbry committed:
ci: split ci.yml into ci-dev.yml ci-rc.yml

Development builds and release candidates require different configuration for testing and deployment.

akomii commented on .github/workflows/ci-dev.yml:
```diff
@@ -0,0 +1,41 @@
+          sudo service wildfly restart
+
+  #      - name: Test
+  #        run:  # TODO
```
I would remove this for now. The job will only do build and deploy. Continous testing will be added at a later time

akomii commented on .github/workflows/ci-dev.yml:
```diff
@@ -0,0 +1,41 @@
+    with:
+      package-name: 'aktin-notaufnahme-i2b2'
+      package-version: ${{ needs.build.outputs.package-version }}
+      codename: 'testing'
```
The release type should be included in the codename, so "jammy-testing" is the correct name. But as the name of the job is ci-dev, maybe "jammy-devel" would be more appropriate? Its up to you...

bunbry replied:
`testing` feels appropriate for builds which are deployed untested and are in the process of being (manually) tested. `dev` or `development` feels more neutral regarding the test status of the builds (either untested or some automated tests were perform before deployment, depending on the context of the development process).
I would suggest using `jammy-dev`.

bunbry replied:
I just looked into the naming conventions for the codenames, and it looks like we have **`unstable`** and **`testing`** mixed up (see this [FAQ](https://www.debian.org/doc/manuals/debian-faq/ftparchives.en.html), Sections 6.1 and 6.4 - 6.6 in particular).
Our **`testing`** distribution should be named **`unstable`** and vice versa.

akomii replied:
Yes, this will be done

akomii commented on .github/workflows/ci-rc.yml:
```diff
@@ -0,0 +1,38 @@
+          sudo service wildfly restart
+
+  #      - name: Test
+  #        run:  # TODO
```
Again, I would cut the tests for now and add them later because they are a task in themselves and be fully fleshed out. Otherwise the tests will just take a long time to complete (because of the installation and configuration) and offer no additional value

akomii commented on .github/workflows/ci-rc.yml:
```diff
@@ -0,0 +1,38 @@
+    with:
+      package-name: 'aktin-notaufnahme-i2b2'
+      package-version: ${{ needs.build.outputs.package-version }}
+      codename: 'unstable'
```
Should be "jammy-unstable"

bunbry replied:
Same as this https://github.com/aktin/debian-i2b2-pkg/pull/17#discussion_r1927179224

bunbry committed:
revert(ci): remove test jobs

Testing will be implemented in the future.
Beside the completely reverted commit referenced in the footer, 2806c222 is also affected indirectly.

Refs: f77b1f94

bunbry committed:
fix(ci): change target codenames

Target codenames are not simply "jammy" or "unstable" but combinations
like "jammy-unstable" and "jammy-testing".
Debian naming convention uses "testing" for release candidates while
"unstable" is used for the main development branch.

bunbry committed:
ci: remove pull_request trigger

The trigger works different from expectation (triggers on every change).
This trigger might be re-added later.

akomii commented on .github/workflows/ci-rc.yml:
```diff
@@ -0,0 +1,20 @@
+    with:
+      package-name: 'aktin-notaufnahme-i2b2'
+      package-version: ${{ needs.build.outputs.package-version }}
+      codename: 'jammy-testing'
```
To make things simple: I would remove `ci-dev.yml` and only keep the deployment of tagged commits directly on testing. An unstable development branch can be added in a later iteration

bunbry committed:
revert(ci): remove ci-dev.yml

Scope of CI is narrowed to release candidates only.

Refs: 2806c222
